File: PRACTICA 5/EcuacionCuadratica-Modular.py
import math
def getDiscriminante(a, b, c):
    return (b ** 2) - (4 * a * c)
def getRaiz1(a, b, c):
    discriminante = getDiscriminante(a, b, c)
    if discriminante < 0:
        return None  
    return (-b + math.sqrt(discriminante)) / (2 * a)
def getRaiz2(a, b, c):
    discriminante = getDiscriminante(a, b, c)
    if discriminante < 0:
        return None  
    return (-b - math.sqrt(discriminante)) / (2 * a)
def main():
    a, b, c = map(float, input("Ingrese a, b, c: ").split())
    discriminante = getDiscriminante(a, b, c)
    if discriminante > 0:
        print(f"La ecuacion tiene dos raices {getRaiz1(a, b, c):.5f} y {getRaiz2(a, b,    
              c):.5f}")
    elif discriminante == 0:
        print(f"La ecuacion tiene una raiz {getRaiz1(a, b, c):.5f}")
    else:
        print("La ecuacion no tiene raices reales")
if __name__ == "__main__":
    main()

File: PRACTICA 5/JAVA/EcuacionCuadratica-Modular.java
import java.util.Scanner;
/**Maya Salomé Machaca Arcani**/

public class EcuacionCuadratica {
    public static double getDiscriminante(double a, double b, double c) {
        return (b * b) - (4 * a * c);
    }
        public static double getRaiz1(double a, double b, double c) {
        double discriminante = getDiscriminante(a, b, c);
        if (discriminante < 0) return Double.NaN; 
        return (-b + Math.sqrt(discriminante)) / (2 * a);
    }
        public static double getRaiz2(double a, double b, double c) {
        double discriminante = getDiscriminante(a, b, c);
        if (discriminante < 0) return Double.NaN;
        return (-b - Math.sqrt(discriminante)) / (2 * a);
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese a, b, c: ");
        double a = scanner.nextDouble();
        double b = scanner.nextDouble();
        double c = scanner.nextDouble();

        double discriminante = getDiscriminante(a, b, c);

        if (discriminante > 0) {
            System.out.println("La ecuacion tiene dos raices " + getRaiz1(a, b, c) + " y " + 
                               getRaiz2(a, b, c));
        } else if (discriminante == 0) {
            System.out.println("La ecuacion tiene una raz " + getRaiz1(a, b, c));
        } else {
            System.out.println("La ecuacion no tiene raices reales");
        }

        scanner.close();
    }
}

